software/contrib/pams.py

```diff
@@ -0,0 +1,1118 @@
+            self.last_interaction_time = now
+
+
+        @b2.handler_falling
+        def on_b2_release():
+            """Handler for releasing button 2
+
+            Handle long vs short presses differently
+
+            Button 2 is used to cycle between screens
+
+            If the screensaver is visible, just wake up the display & don't process
+            the actual button click/long-press
+            """
+            now = time.ticks_ms()
+            if time.ticks_diff(now, self.last_interaction_time) <= SCREENSAVER_TIMEOUT_MS:
+                if time.ticks_diff(now, b2.last_pressed()) > LONG_PRESS_MS:
+                    # long press
+                    # change between the main & sub menus
+                    self.main_menu.on_long_press()
+                else:
+                    # short press
+                    self.main_menu.on_click()
+                    self.save()
+
+            self.last_interaction_time = now
+
+    def load(self):
+        """Load parameters from persistent storage and apply them
+        """
+        try:
+            state = self.load_state_json()
+
+            channel_cfgs = state.get("channels", [])
+            for i in range(len(channel_cfgs)):
+                self.channels[i].load_settings(channel_cfgs[i])
+
+            clock_cfg = state.get("clock", None)
+            if clock_cfg:
+                self.clock.load_settings(clock_cfg)
+
+            din_cfg = state.get("din", None)
+            if din_cfg:
+                self.din_mode.load(din_cfg)
+
+            ain_cfg = state.get("ain", [])
+            cv_keys = list(CV_INS.keys())
+            for i in range(len(ain_cfg)):
+                CV_INS[cv_keys[i]].load_settings(ain_cfg[i])
+
+        except:
+            print("[ERR ] Error loading saved configuration for PamsWorkout")
+            print("[ERR ] Please delete the storage file and restart the module")
+
+    def save(self):
+        """Save current settings to the persistent storage
+        """
+        state = {
+            "clock": self.clock.to_dict(),
+            "channels": [],
+            "din": self.din_mode.to_dict(),
+            "ain": []
+        }
+        for i in range(len(self.channels)):
+            state["channels"].append(self.channels[i].to_dict())
+        for cv in CV_INS.keys():
+            state["ain"].append(CV_INS[cv].to_dict())
+
+        self.save_state_json(state)
+
+    @classmethod
+    def display_name(cls):
+        return "Pam's Workout"
+
+    def main(self):
+        splash = SplashScreen()
+        splash.draw()
+
+        self.load()
+
+        time.sleep(0.5)
+
+        while True:
+            now = time.ticks_ms()
+
+            for cv in CV_INS.values():
+                cv.update()
+
+            if time.ticks_diff(now, self.last_interaction_time) > BLANK_TIMEOUT_MS:
+                self.screensaver.draw_blank()
+            elif time.ticks_diff(now, self.last_interaction_time) > SCREENSAVER_TIMEOUT_MS:
```

Comment:
[optional] It might be worth throwing the calculated elapsed time into a local variable. This would avoid calculating it twice, as well as make it clear that you are using the same duration in both branches of the `if`.